File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        // sort(nums), 
        // count the longest consecutive seq
        // O(nlogn)

        // consecutive seq => the longest = nums.length
        // longest: minimum of nums: mini + nums.length = maxi
        // only need to check from mini of nums to mini + nums.length
        // O(n)
        // traversal the nums, build a lookup set
        // from the minimal, we check the longest consectutive seq
        if (nums.length <= 1 ){
            return nums.length;
        }
        int ans = 0;
 
        HashSet<Integer> lookup = new HashSet<>();
       
        for (int num:nums){
            lookup.add(num);
        }

        for (int key:lookup){
            System.out.println(key);
            int curr = key;
            if (lookup.contains(curr -1)){
                continue;
            }
            int count = 1;
            while (lookup.contains(curr + 1)){
                curr += 1;
                count += 1;
            }
            ans = Math.max(count,ans);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()

        for a in range(len(nums) - 2):
            l = a + 1
            r = len(nums) - 1
            while l < r:
                sumi = nums[l] + nums[r] + nums[a]
                if sumi == 0:
                    curr = [nums[a],nums[l],nums[r]]
                    if curr not in ans:
                        ans.append([nums[a],nums[l],nums[r]])
                    l += 1
                elif sumi < 0:
                    l += 1
                else:
                    r -= 1
        
        return ans